### PR TITLE
Support "register schema only" by set LOOP=0

### PR DESCRIPTION
**Pull request by l2280212**
Fix the Out Of Memory issue in Benchmark caused by creating a large number of devices using templates.

### CODE REVIEW

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/BaseMode.java [hunk 1]
```diff
@@ -62,7 +62,17 @@ public abstract class BaseMode {
...
+    List<DBConfig> dbConfigs = new ArrayList<>();
+    dbConfigs.add(config.getDbConfig());
+    if (config.isIS_DOUBLE_WRITE()) {
+      dbConfigs.add(config.getANOTHER_DBConfig());
+    }
+    if (config.isIS_DELETE_DATA() && (!cleanUpData(dbConfigs, measurement))) {
+      return false;
+    }
+    return !config.isCREATE_SCHEMA() || (registerSchema(measurement));
+  }
```
Comment (SpriCoder) [inline core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/BaseMode.java:75]: Consider whether these options need to be done before all mode

File: iotdb-1.1/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb110/IoTDB.java [hunk 2]
```diff
@@ -305,18 +305,14 @@ private void registerStorageGroups(Session metaSession, List<TimeseriesSchema> s
...
-              }
-            });
+    for (TimeseriesSchema timeseriesSchema : schemaList) {
+      someDevicePaths.add(timeseriesSchema.getDeviceId());
+      if (someDevicePaths.size() >= ACTIVATE_TEMPLATE_THRESHOLD) {
+        activateTemplateForSomeDevices(metaSession, someDevicePaths, activatedDeviceCount.get());
+        activatedDeviceCount.addAndGet(someDevicePaths.size());
+        someDevicePaths.clear();
+      }
+    }
```
Comment (SpriCoder) [inline iotdb-1.1/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb110/IoTDB.java:315]: Please check `map` operations, I think this change may cause some problem

**Review by SpriCoder — COMMENTED** [review]
Good Job! But there are some problems PTAL

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java [hunk 3]
```diff
@@ -621,14 +621,7 @@ private boolean checkOperationProportion() {
...
     }
-    String[] op = config.getOPERATION_PROPORTION().split(":");
-    int minOps = 0;
-    for (String s : op) {
-      if (Double.parseDouble(s) > 1e-7) {
-        minOps++;
-      }
-    }
-    if (minOps > config.getLOOP()) {
+    if (config.getLOOP() < 0) {
```
[the remark this answers is not in the document]
Comment (l2280212) [reply core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java:624]: I have checked the device names created by template
image

File: iotdb-1.1/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb110/IoTDB.java (hunk 2 above)
Comment (l2280212) [reply iotdb-1.1/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb110/IoTDB.java:315]: I have checked the device names created by template
image

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (SpriCoder) [reply core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java:624]: Oh, I mean that we need to check OPERATION_PROPORTION here at the same time

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/BaseMode.java (hunk 1 above)
Comment (l2280212) [reply core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/BaseMode.java:75]: I am also confused about whether to make modifications here. Two classes extend base class and  use nearly identical code for `preCheck`, while the other two employ different approaches for `preCheck`. I can revert this change

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/VerificationWriteMode.java [hunk 4]
```diff
@@ -20,37 +20,17 @@
...
-import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 public class VerificationWriteMode extends BaseMode {
 
-  private static final Config config = ConfigDescriptor.getInstance().getConfig();
-
   @Override
   protected boolean preCheck() {
```
Comment (liyuheng55555) [inline core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/VerificationWriteMode.java:32]: seems not necessary to call super.preCheck()

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/TestWithDefaultPathMode.java [hunk 5]
```diff
@@ -39,23 +38,7 @@ protected boolean preCheck() {
     PersistenceFactory persistenceFactory = new PersistenceFactory();
```
Comment (liyuheng55555) [inline core/src/main/java/cn/edu/tsinghua/iot/benchmark/mode/TestWithDefaultPathMode.java:38]: Maybe move these 3 lines to constructor ?

File: core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java [hunk 6]
```diff
@@ -621,14 +621,7 @@ private boolean checkOperationProportion() {
         != config.getOPERATION_PROPORTION_LEN()) {
       config.setOPERATION_PROPORTION(config.getOPERATION_PROPORTION() + ":0");
     }
-    String[] op = config.getOPERATION_PROPORTION().split(":");
```
Comment (liyuheng55555) [inline core/src/main/java/cn/edu/tsinghua/iot/benchmark/conf/ConfigDescriptor.java:624]: How about, we remain this check, but LOGGER.warn something then return true ?

**Review by liyuheng55555 — CHANGES_REQUESTED** [review]
Please take a look~

**Review by SpriCoder — APPROVED** [review]
LGTM~

**Review by liyuheng55555 — APPROVED** [review]
LGTM~ Good work !

**Pull request merged**